Missing compiler error on optionals

**V version:**
V 0.1.18 (recent git commit: `7fc678c 2019-09-09 fix urllib_test.v (HEAD -> master, origin/master, origin/HEAD) [Alexander Medvednikov]|`)

**OS:** 
Linux t500 4.19.67-1-lts #1 SMP Fri Aug 16 10:30:45 CEST 2019 x86_64 GNU/Linux

**What did you do?**
Treating a function without *optional return value* like a function with *optional return value* is not caught by *v* compiler as error.
```go
fn f() int {
  return 7
}
fn main(){
  x := f() or { panic( err ) }
  println( x )
}
```

**What did you expect to see?**
A readable compiler error, that the function `f()` doesn't return optional value and therefore can't panic.
 
**What did you see instead?**
```
$ ./v -debug test01.v
C compiler=cc
/home/user/del/v/test01.tmp.c: In function ‘main’:
/home/user/del/v/test01.tmp.c:3483:12: error: request for member ‘ok’ in something not a structure or union
 3483 |  if (!tmp2 .ok) {
      |            ^
/home/user/del/v/test01.tmp.c:3484:20: error: request for member ‘error’ in something not a structure or union
 3484 |  string err = tmp2 . error;
      |                    ^
/home/user/del/v/test01.tmp.c:3489:23: error: request for member ‘data’ in something not a structure or union
 3489 |  int x = *(int*) tmp2 . data;
      |                       ^
V error: C error. ...
```

------
P.S. `v -v` shall show the git commit id (preferably with the git commit date & time), otherwise the version info is not enough to distinguish it anywhere :cry:.

--- a/compiler/gen_c.v
+++ b/compiler/gen_c.v
@@ -29,6 +29,9 @@ fn (p mut Parser) gen_var_decl(name string, is_static bool) string {
 		// p.assigned_var = ''
 		p.cgen.set_placeholder(pos, '$typ $tmp = ')
 		p.genln(';')
+		if !typ.starts_with('Option_') {
+			p.error('`or` block cannot be applied to non-optional type')
+		}
 		typ = typ.replace('Option_', '')
 		p.next()
 		p.check(.lcbr)
@@ -61,7 +64,7 @@ fn (p mut Parser) gen_var_decl(name string, is_static bool) string {
 			p.cgen.resetln(' = {' + initializer.all_after('{') )
 		} else if initializer.len == 0 {
 			p.cgen.resetln(' = { 0 }')
-		}	
+		}
 	}
 
 	if is_static {
@@ -196,7 +199,7 @@ fn (p mut Parser) index_get(typ string, fn_ph int, cfg IndexCfg) {
 		// p.cgen.insert_before('if (!${tmp}.str) $tmp = tos("", 0);')
 		p.cgen.insert_before('if (!$tmp_ok) $tmp = tos((byte *)"", 0);')
 	}
-	
+
 }
 
 fn (table mut Table) fn_gen_name(f &Fn) string {
@@ -296,7 +299,7 @@ fn (p mut Parser) gen_array_at(typ_ string, is_arr0 bool, fn_ph int) {
 	else {
 		p.gen(',')
 	}
-}	
+}
 
 fn (p mut Parser) gen_for_header(i, tmp, var_typ, val string) {
 	p.genln('for (int $i = 0; $i < ${tmp}.len; $i++) {')
@@ -345,11 +348,11 @@ fn (p mut Parser) gen_array_init(typ string, no_alloc bool, new_arr_ph int, nr_e
 		p.gen(' })')
 	}
 	// Need to do this in the second pass, otherwise it goes to the very top of the out.c file
-	if !p.first_pass() {		
+	if !p.first_pass() {
 		p.cgen.set_placeholder(new_arr_ph,
 			'$new_arr($nr_elems, $nr_elems, sizeof($typ), EMPTY_ARRAY_OF_ELEMS( $typ, $nr_elems ) { ')
 	}
-}	
+}
 
 fn (p mut Parser) gen_array_set(typ string, is_ptr, is_map bool,fn_ph, assign_pos int, is_cao bool) {
 	// `a[0] = 7`

--- a/compiler/tests/option_test.v
+++ b/compiler/tests/option_test.v
@@ -19,7 +19,7 @@ fn err_call(ok bool) ?int {
 fn ret_none() ?int {
 	//return error('wtf') //none
 	return none
-}	
+}
 
 fn test_option_for_base_type_without_variable() {
 	val := err_call(true) or {
@@ -30,15 +30,19 @@ fn test_option_for_base_type_without_variable() {
 	val2 := ret_none() or {
 		println('yep')
 		return
-	}	
-	println('nice')
-	println(val2)
+	}
+	println('$val2 should have been `none`')
+	assert false
+	// This is invalid:
+	//	x := 5 or {
+	//		return
+	//	}
 }
 
 fn test_if_opt() {
 	if val := err_call(false) {
 		assert val == 42
-	}	
+	}
 	assert 1 == 1
 	println('nice')
-}	
+}

